radix-engine/src/engine/track.rs

```diff
@@ -147,13 +142,46 @@ impl<'s, S: SubstateStore> Track<'s, S> {
         }
     }
 
+    fn get_substate<A: Encode, T: Decode>(&self, address: &A) -> Option<T> {
+        self.substate_store
+            .get_substate(address)
+            .and_then(|v| scrypto_decode(&v).map(|r| Some(r)).unwrap_or(None))
```

Comment:
The `unwrap_or(None)` looks dangerous to me. Should panic instead of swallowing this illegal state. There are a few more places like this.

Reply:
Ah yes, this made more sense when there was only a single type of Address and wasn't sure the type you'd get back.